src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.cs

```diff
@@ -36,5 +42,42 @@ internal DefaultAnalyzerAssemblyLoader(System.Runtime.Loader.AssemblyLoadContext
         /// <param name="fullPath"></param>
         /// <returns></returns>
         protected override string PreparePathToLoad(string fullPath) => fullPath;
+
+        /// <summary>
+        /// Return an <see cref="IAnalyzerAssemblyLoader"/> which does not lock assemblies on disk that is
+        /// most appropriate for the current platform.
+        /// </summary>
+        /// <param name="windowsShadowPath">A shadow copy path will be created on Windows and this value 
+        /// will be the base directory where shadow copy assemblies are stored. </param>
+        internal static IAnalyzerAssemblyLoader CreateNonLockingLoader(string? windowsShadowPath = null)
```

Comment:
The issue with making this required is that every consumer must provide a value on every operating system. Even though tis value is not used on Linux / Mac customers still must provide it. That will push consumers to use either `Path.GetTempPath` which will trigger scrutiny as it's an API to avoid for cases like this or invent a location on disk that we'll never actually use. Basically it makes call sites complicated and error prone. 

The current setup is meant to avoid that. By default the implementation will choose an appropriate and safe location. In the cases where you are comfortable picking a shadow copy path then you are free to do so.

Reply:
Chatted offline and decided to make it mandatory.